Snakefile

```diff
@@ -275,6 +275,23 @@ PARALLEL_LANGUAGE_PAIRS = [
     'en_ru', 'en_sv', 'en_zh-Hans', 'en_zh-Hant'
 ]
 
+# We want some flexibility in the amount of data we encode via SentencePiece
+# to use in training a language model.  That data will be a subset of the
+# monolingual data sampled to train SentencePiece itself; we define the size
+# of that subset here as the reciprocal of the fraction of the whole monolingual
+# dataset to use.  (So setting this to, say, 5 uses one part in 5 of the data.)
+RECIPROCAL_OF_FRACTION_OF_DATA_TO_SPM_ENCODE = 10
+
+# To have a (statically) predictable set of SentencePiece id files when
+# segregated by length (number of id's per input, not the number of inputs),
+# we set an upper bound on the length.  Sentences with longer encodings will
+# be split into chunks with sizes ranging between two specified lengths.
+# The maximum possible length is set based on Salesforce's awd-lstm-lm code.
+MAX_SPM_ENCODE_IDS = 70
+MIN_SPM_CHUNK_LEN = 15
+MAX_SPM_CHUNK_LEN = 30
```

Comment:
It would work fine for `MAX_SPM_CHUNK_LEN` to be 70 (and MIN to correspondingly be 35), right? I suggested 15 and 30 before you showed me the sequence length that AWD-LSTM uses.

Reply:
Yes, those values should be fine; I'll make the changes.  We can tweak these values again later if future experience so dictates.